dotcom-rendering/docs/architecture/027-pictures.md

```diff
@@ -0,0 +1,133 @@
+
+For example imagine a phone with a super high resolution 1200px wide (2400px high) screen. Following our breakpoint sizes, we'd try and render a desktop type experience for this user. However, the reality is this screen is only maybe 5 inches across, so the site would be totally unusable.
+
+CSS Pixels & DPR to the rescue! Our browser can use a different width for calculating breakpoints, media queries, etc than the real resolution of the screen. This is CSS pixels. So let's say for the sake of argument the browser uses 300px for our CSS pixel width - Wohoo, we're displaying a mobile experience, all is well. The DPR is the ratio between CSS pixels and actual resolution, so 1200 / 300 gives us a DPR of 4. Why this is important will be discussed later.
+
+#### The DPR Problem
+
+This problem comes from how the browser tries to compensate for high DPR displays when choosing an image source. In previous iterations of Frontend & DCR, we provided only 2 sources - a regular set, and a set of sources for high DPR displays; targeted with a media query to ensure it's picked.
+
+Unfortunately the browser itself would try to compensate for high DPR as well, but in a less efficient way. Once the browser had figured out what size source it wants from the `sizes` attribute on our source element, it's then multiplied by the DPR of the display to get the desired width it will look for in our `srcset`:
+
+```
+# desiredWidth is the width of an image that the browser will look for in `srcset`
+# size is the chosen size based off our queries in the `sizes` attribute
+# DPR is the ration between CSS Pixels & device resolution (e.g 2, 3, 4)
+desiredWidth = size * DPR
+```
+
+This posed a problem for us, because if we tell the browser "Hey choose a 620px image", and the browser has a DPR of say 3, it will actually look for an image source for 1860px, far higher resolution than is needed for the user to have a good experience.
+
+#### The DPR Solution
+
+This problem was first solved in Frontend, then replicated in DCR
+
+Rather than having just 2 source elements, we instead have 2 source elements per breakpoint (one for high DPR, one for regular displays), and provide just one size & source set for that source element.
+
+Lets look at a simplified example (with only 1 source per breakpoint):
+
+```html
+<picture>
+    <source
+        media="min-width: 980px"
+        sizes="620px"
+        srcset="https://xxx.png?width=620px 620w"
+    />
+    <source
+        media="min-width: 660px"
+        sizes="620px"
+        srcset="https://xxx.png?width=620px 620w"
+    />
+    <source
+        media="min-width: 480px"
+        sizes="480px"
+        srcset="https://xxx.png?width=480px 480w"
+    />
+    <source
+        media="min-width: 375px"
+        sizes="420px"
+        srcset="https://xxx.png?width=420px 420w"
+    />
+</picture>
```

Comment:
Having a `sizes` attribute if you have only one element in your `scrset` is redundant, and maybe a little confusing.

```suggestion
<picture>
    <source
        media="min-width: 980px"
        srcset="https://xxx.png?width=620px 620w"
    />
    <source
        media="min-width: 660px"
        srcset="https://xxx.png?width=620px 620w"
    />
    <source
        media="min-width: 480px"
        srcset="https://xxx.png?width=480px 480w"
    />
    <source
        media="min-width: 375px"
        srcset="https://xxx.png?width=420px 420w"
    />
</picture>
```

Reply:
Thanks! I have sorted this now :) Sizes begone!